### PR TITLE
Workflow layout

### PR DESCRIPTION
**Pull request by chriswhong**
- First round of layout and styles for the workflow editor.
- Updates package.json to fix some dependency errors that happened after moving the frontend code up to root.
- Cleans up some unused imports
workflow

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -103,5 +103,8 @@
     "@storybook/react": "^6.1.15",
     "@types/react-router-dom": "^5.1.6",
     "ts-jest": "^26.4.4"
+  },
+  "resolutions": {
+    "babel-loader": "8.1.0"
```
Comment (b5) [inline package.json:108]: I ran into this issue, seems to me the resolver isn't needed with the react storybook upgrade. Mind upgrading react storybook & trying to remove this & seeing if `yarn && yarn start ` works?

File: src/features/dataset/DatasetMiniHeader.tsx [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+import Button from '../../chrome/Button'
+import TextLink from '../../chrome/TextLink'
+import { Dataset, qriRefFromDataset } from '../../qri/dataset'
+import Icon from '../../chrome/Icon'
+import DownloadDatasetButton from '../download/DownloadDatasetButton'
+
+export interface DatasetHeaderProps {
+  dataset: Dataset
+  show: boolean
```
Comment (b5) [inline src/features/dataset/DatasetMiniHeader.tsx]: Flip the logic for this, components should display by default. I'd change the name to `hide`

File: src/features/workflow/Block.tsx [hunk 3]
```diff
@@ -2,17 +2,16 @@ import React from 'react'
...
-  description: string
   onClick?: () => void
 }
 
-const Block: React.FC<any> = ({ name, description, onClick }) => (
-  <div className='my-2 px-2' onClick={onClick}>
-    <div className='bg-gray-100 px-4 py-2 rounded mt-1 border-b border-gray-300 cursor-pointer hover:bg-gray-100'>
-      <div className='font-semibold pb-1'>{name}</div>
-      <div className='text-xs'>{description}</div>
+const Block: React.FC<any> = ({ name, children,onClick }) => (
```
Comment (b5) [inline src/features/workflow/Block.tsx]: Any -> BlockProps

File: src/features/workflow/WorkflowOutline.tsx [hunk 4]
```diff
@@ -24,59 +25,61 @@ const WorkflowOutline: React.FC<WorkflowOutlineProps> = ({
...
           </div>
-        </ScrollTrigger>}
-      </div>
-      <ScrollTrigger target='on-completion'><div className='font-semibold text-gray-900 mb-2 uppercase text-xs tracking-wide'>On Completion</div></ScrollTrigger>
-      <div className='text-xs text-gray-500 bg-gray-200 inline-block py-0 px-2 rounded-xl border ml-2'>
-        <Icon icon='cloudUpload' size='sm' className='mr-1'/> <span>push to cloud</span>
+          <ScrollTrigger target='on-completion'><div className='font-semibold text-qrinavy mb-2'>On Completion</div></ScrollTrigger>
+
+          <div className="mb-20">
+            {workflow && <DeployButtonWithStatusDescription workflow={workflow} />}
```
Comment (b5) [inline src/features/workflow/WorkflowOutline.tsx:77]: Still not a fan of having the deploy button on the sidebar, but let's take that debate up when chatting with the rest of product squad. Fine to leave this in for now

File: src/features/workflow/WorkflowPage.tsx [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+import NavBar from '../navbar/NavBar'
+import DatasetNavSidebar from '../dataset/DatasetNavSidebar'
+import DeployingScreen from '../deploy/DeployingScreen'
+import { selectSessionUserCanEditDataset, selectDataset } from '../dataset/state/datasetState'
+import { QriRef } from '../../qri/ref'
+import Workflow from './Workflow'
+import Scroller from '../scroller/Scroller'
+
+
+interface DatasetPreviewPageProps {
```
Comment (b5) [inline src/features/workflow/WorkflowPage.tsx]: Rename component to match file

**Review by b5 — APPROVED** [review]
I should review more carefully, but this looks good so far. Considering you want to build on top of these changes I vote we clean up these Nita, merge, and deal with any bugs introduced in follow ups

File: package.json (hunk 1 above)
Comment (chriswhong) [reply package.json:108]: No luck upgrading storybook, we get the same error.

Comment (chriswhong) [reply package.json:108]: https://github.com/storybookjs/storybook/issues/5183

Here's a very long issue, but there's no real fix yet.  The workarounds are to add `resolutions` to package.json as we have done here, or just to silence the warning.

**Pull request merged**